### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by sheylaPozo**
# Processing data in models
**Models:**

✔ Create model classes for all entities as shown in the ERD diagram.
✔ Set up associations between models.

### **Include some additional methods:**

**User:**
✔ A method that returns the 3 most recent posts for a given user.
**Post:**
✔A method that updates the posts counter for a user.
✔A method that returns the 5 most recent comments for a given post.
**Comment:**
✔A method that updates the comments counter for a post.
**Like:**
✔A method that updates the likes counter for a post.
**ApplicationController:**
✔Create a current_user method in ApplicationController, that returns the first user from a database.

Thank you for kindly reviewing my project ❄😋

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
...
-  has_many :posts
-  has_many :likes
-  has_many :comments
+  has_many :comments, dependent: :destroy
+  has_many :posts, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def recent_posts
+    posts.last(3)
+  end
```
Comment (DeliceLydia) [inline app/models/user.rb]: - [x] I suggest that you make a DESC order you can do like `posts.order('created_at Desc').limit(3)`

File: app/models/post.rb [hunk 2]
```diff
@@ -1,5 +1,15 @@
 class Post < ApplicationRecord
   belongs_to :user
-  has_many :likes
-  has_many :comments
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  after_create :update_counter
```
Comment (DeliceLydia) [inline app/models/post.rb]: ## Optional
- [x] I suggest that you use [Rails active callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` to call the `update_counter` method every time a new post has been saved to the database, then make the `update_counter` method private

File: app/models/like.rb [hunk 3]
```diff
@@ -1,4 +1,10 @@
 class Like < ApplicationRecord
   belongs_to :user
   belongs_to :post
+
+  after_create :update_counter
+
+  def update_counter
+    post.update(likes_counter: post.likes.size)
+  end
```
Comment (DeliceLydia) [inline app/models/like.rb]: ## Optional
- [x] Same here you can use the `after_save` rails callback

File: app/models/comment.rb [hunk 4]
```diff
@@ -1,4 +1,10 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
+
+  after_create :update_counter
+
+  def update_counter
+    post.update(comments_counter: post.comments.size)
+  end
```
Comment (DeliceLydia) [inline app/models/comment.rb]: ## Optional
- [x] Same here I suggest that you use the `after_save` and remember to make `update_counter` private

File: app/models/post.rb [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
+
+  def update_counter
+    user.update(posts_counter: user.posts.size)
+  end
+
+  def recent_comments
+    comments.last(5)
+  end
```
Comment (DeliceLydia) [inline app/models/post.rb]: - [x] I suggest that comments be DESC order as well like `comments.order('created_at Desc').limit(5)`

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @sheylaPozo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Davidosky007** [comment]
## Hi @sheylaPozo 👋, 

## Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 


done-gif

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sheylaPozo** [comment]
thank you :)

**Pull request merged**